src/Sentry/SentrySdk.cs

```diff
@@ -49,6 +49,19 @@ internal static IHub InitHub(SentryOptions options)
             options.LogWarning("The provided DSN that contains a secret key. This is not required and will be ignored.");
         }
 
+        if (AotHelper.IsNativeAot)
+#pragma warning disable CS0162 // Unreachable code detected
+        {
+#pragma warning disable 0162 // Unreachable code on old .NET frameworks
+            options.LogDebug("This looks like a Native AOT application build.");
+#pragma warning restore 0162
+        }
+        else
+        {
+#pragma warning restore CS0162 // Unreachable code detected
+            options.LogDebug("This doesn't look like a Native AOT application build.");
```

Comment:
I agree with the `// TODO this probably more closely represents trimming rather than NativeAOT?`. Enabling AOT compilation and then reading this in the logs does seem odd.